File: internal/repository/settings_repo.go
package repository

import "Rocket-Invo/internal/database"

type SettingsRepo struct {
	db *database.DB
}

func NewSettingsRepo(db *database.DB) *SettingsRepo {
	return &SettingsRepo{db}
}

func (r *SettingsRepo) Get(key string) (string, error) {
	var value string
	err := r.db.QueryRow("SELECT value FROM settings WHERE key=?", key).Scan(&value)
	if err != nil {
		return "", err
	}
	return value, nil
}

func (r *SettingsRepo) Set(key, value string) error {
	_, err := r.db.Exec("INSERT OR REPLACE INTO settings (key, value) VALUES (?, ?)", key, value)
	return err
}

func (r *SettingsRepo) GetAll() (map[string]string, error) {
	rows, err := r.db.Query("SELECT key, value FROM settings")
	if err != nil {
		return nil, err
	}
	defer rows.Close()

	settings := make(map[string]string)
	for rows.Next() {
		var key, value string
		if err := rows.Scan(&key, &value); err != nil {
			return nil, err
		}
		settings[key] = value
	}
	return settings, rows.Err()
}

File: electron/database/settings-repo.ts
import { Database as SqlJsDatabase } from 'sql.js'
import { get, run, all } from './query'
import { SettingsMap } from '../types'

export class SettingsRepo {
  private db: SqlJsDatabase

  constructor(db: SqlJsDatabase) {
    this.db = db
  }

  get(key: string): string | undefined {
    const row = get<{ value: string }>(this.db, 'SELECT value FROM settings WHERE key = ?', [key])
    return row?.value
  }

  set(key: string, value: string): void {
    run(this.db, 'INSERT OR REPLACE INTO settings (key, value) VALUES (?, ?)', [key, value])
  }

  getAll(): SettingsMap {
    const rows = all<{ key: string; value: string }>(this.db, 'SELECT key, value FROM settings')
    const result: SettingsMap = {}
    for (const row of rows) {
      result[row.key] = row.value
    }
    return result
  }
}
